src/parser/jobs/mnk/RiddleOfFire.js

```diff
@@ -0,0 +1,160 @@
+import React, {Fragment} from 'react'
+import {Accordion} from 'semantic-ui-react'
+
+import {ActionLink, StatusLink} from 'components/ui/DbLink'
+import Rotation from 'components/ui/Rotation'
+import ACTIONS, {getAction} from 'data/ACTIONS'
+import STATUSES from 'data/STATUSES'
+import Module from 'parser/core/Module'
+import {Suggestion, TieredSuggestion, SEVERITY} from 'parser/core/modules/Suggestions'
+import {matchClosestHigher} from 'utilities'
+
+const ROF_DURATION = STATUSES.RIDDLE_OF_FIRE.duration * 1000
+
+const POSSIBLE_GCDS = 9
+
+export default class RiddleOfFire extends Module {
+	static handle = 'riddleoffire'
+	static dependencies = [
+		'suggestions',
+	]
+
+	static title = 'Riddle of Fire'
+
+	_active = false
+	_history = []
+	_riddle = {}
+	_rushing = false
+
+	_missedGcds = 0
+	_missedTks = 0
+
+	constructor(...args) {
+		super(...args)
+		this.addHook('cast', {by: 'player'}, this._onCast)
+		this.addHook('removebuff', {by: 'player', abilityId: STATUSES.RIDDLE_OF_FIRE.id}, this._onDrop)
+		this.addHook('complete', this._onComplete)
+	}
+
+	_onCast(event) {
+		const actionId = event.ability.guid
+
+		if (actionId === ACTIONS.RIDDLE_OF_FIRE.id) {
+			this._active = true
+			this._riddle = {
+				start: event.timestamp,
+				end: null,
+				casts: [],
+			}
+
+			const fightTimeRemaining = this.parser.fight.end_time - event.timestamp
+			this._rushing = ROF_DURATION >= fightTimeRemaining
+		}
+
+		// we only care about actual skills
+		if (!this._active || getAction(actionId).autoAttack) {
+			return
+		}
+
+		this._riddle.casts.push(event)
+	}
+
+	_onDrop(event) {
+		this._stopAndSave(event.timestamp)
+	}
+
+	_onComplete() {
+		// Close up if RoF was active at the end of the fight
+		if (this._active) {
+			this._stopAndSave()
+		}
+
+		// Aggregate GCDs under each RoF
+		const rofs = []
+		this._history.forEach(riddle => {
+			rofs.push(riddle.casts.filter(cast => getAction(cast.ability.guid).onGcd).length)
+		})
+
+		if (this._missedGcds > 0) {
+			this.suggestions.add(new TieredSuggestion({
+				icon: ACTIONS.RIDDLE_OF_FIRE.icon,
+				severity: SEVERITY.MAJOR,
```

Comment:
You can omit this if it's a `TieredSuggestion`.

Reply:
Typo, part way through changing it to Tiered and left some old crap in, oops.

Reply:
I figured as much, just wanted to make sure it got cleaned up. :blobmorning: